unity-ggjj/Assets/Scripts/NarrativeScript/NarrativeScriptPlayer/StoryPlayer.cs

```diff
@@ -0,0 +1,214 @@
+    }
+
+    /// <summary>
+    /// Checks if a story can continue, and handles what happens
+    /// it cannot, depending on the current GameMode
+    /// </summary>
+    /// <returns>If the story can continue (true) or (not)</returns>
+    private bool HandleCannotContinue()
+    {
+        if (Story.canContinue)
+        {
+            return false;
+        }
+        
+        if (!IsAtChoice)
+        {
+            _parent?.EndSubStory();
+            return true;
+        }
+
+        switch (GameMode)
+        {
+            case GameMode.Dialogue:
+                _choiceMenu.Initialise(Story.currentChoices);
+                break;
+            case GameMode.CrossExamination:
+                HandleChoice(0);
+                break;
+            default:
+                throw new InvalidEnumArgumentException($"{GameMode} is an invalid GameMode");
+        }
+
+        return true;
+    }
+
+    /// <summary>
+    /// Selects a choice index in the Ink story and continues the story
+    /// </summary>
+    /// <param name="choiceIndex">The index of the choice to choose</param>
+    public void HandleChoice(int choiceIndex)
+    {
+        Story.ChooseChoiceIndex(choiceIndex);
+        Continue();
+    }
+
+    /// <summary>
+    /// Starts a sub-story which will run in place of the parent story until it ends or is stopped
+    /// </summary>
+    /// <param name="narrativeScript"></param>
+    public void StartSubStory(NarrativeScript narrativeScript)
+    {
+        _subStory = new StoryPlayer(_narrativeScriptPlaylist, _appearingDialogueController, _actionDecoder, _choiceMenu)
+        {
+            ActiveNarrativeScript = narrativeScript,
+            _parent = this
+        };
+        _appearingDialogueController.StopPrintingText();
+        _subStory.Continue(true);
+    }
+
+    /// <summary>
+    /// Ends the current sub-story
+    /// </summary>
+    private void EndSubStory()
+    {
+        _subStory = null;
+        Continue();
+    }
+
+    /// <summary>
+    /// Handles presenting of evidence.
+    /// Checks if evidence is required, then checks if the
+    /// given evidence is correct depending on the Ink story
+    /// </summary>
+    /// <param name="courtRecordObject">The court record object that is required to present</param>
+    public void PresentEvidence(ICourtRecordObject courtRecordObject)
+    {
+        if (_subStory != null)
+        {
+            _subStory.PresentEvidence(courtRecordObject);
+            return;
+        }
+
+        if (!IsAtChoice || GameMode != GameMode.CrossExamination)
+        {
+            return;
+        }
+
+        var currentChoices = Story.currentChoices;
+        var choice = currentChoices.FirstOrDefault(choice => choice.text == courtRecordObject.InstanceName);
+        if (choice != null)
+        {
+            HandleChoice(choice.index);
+        }
+        else
+        {
+            StartSubStory(_narrativeScriptPlaylist.GetRandomFailureScript());
+        }
```

Comment:
Nit: You know me... 👀
```suggestion
        if (choice == null)
        {
            StartSubStory(_narrativeScriptPlaylist.GetRandomFailureScript());
            return;
        }
        
        HandleChoice(choice.index);
        
```